module/VuFind/src/VuFind/Exception/Mail.php

```diff
@@ -40,4 +40,66 @@
  */
 class Mail extends \Exception
 {
+    /**
+     * Default error message when the error is not known exactly.
+     *
+     * @var int
+     */
+    public const ERROR_UNKNOWN = 0;
+
+    /**
+     * Mail recipient address is invalid.
+     *
+     * @var int
+     */
+    public const ERROR_INVALID_RECIPIENT = 1;
+
+    /**
+     * Mail sender address is invalid.
+     *
+     * @var int
+     */
+    public const ERROR_INVALID_SENDER = 2;
+
+    /**
+     * Mail reply to address is invalid.
+     *
+     * @var int
+     */
+    public const ERROR_INVALID_REPLY_TO = 3;
+
+    /**
+     * Mail too many recipients.
+     *
+     * @var int
+     */
+    public const ERROR_TOO_MANY_RECIPIENTS = 4;
+
+    /**
+     * SMS unknown carrier.
+     *
+     * @var int
+     */
+    public const ERROR_UNKNOWN_CARRIER = 5;
+
+    /**
+     * Response unknown.
+     *
+     * @var int
+     */
+    public const ERROR_RESPONSE_UNKNOWN = 6;
+
+    /**
+     * Returns the error message, but excludes too technical messages.
+     *
+     * @return string
+     */
+    public function getDisplayMessage(): string
+    {
+        // If application env is development, we can display too technical messages
+        if ('development' === APPLICATION_ENV || $this->getCode() !== self::ERROR_UNKNOWN) {
+            return $this->getMessage();
+        }
+        return 'email_failure';
```

Comment:
It still troubles me a bit that when sending SMS, we'll display an email failure message by default. I think perhaps we need a different exception class for SMS. What if we change the default display message so that it's a property -- `i.e. protected $defaultDisplayMessage = 'email_failure';` and then we can create a new `VuFind\Exception\SMS` as a subclass of `VuFind\Exception\Mail` which overrides the defaultDisplayMessage to `'Problem sending text.'`. Then the SMS code can throw more specific exceptions, and everything else can keep working as it is.

Am I overthinking this? Maybe that's too much work for a minor point. But let me know if you think it's worth doing!

Reply:
Well, I think it would be ok and it should not be too much of a hazzle. I can add that also.